openmls/src/treesync/node/leaf_node.rs

```diff
@@ -114,6 +206,26 @@ impl LeafNode {
         Ok((leaf_node, encryption_key_pair))
     }
 
+    /// Create a new placeholder [`LeafNode`].
+    ///
+    /// Note: This is not a valid leaf node and it must be rekeyed and signed
+    /// before it can be used.
+    pub(crate) fn new_placeholder(credential_with_key: CredentialWithKey) -> Self {
```

Comment:
This seems a bit hacky. Is there now way to get around the need for a placeholder?

Reply:
As discussed on Zulip, there's no easy way to get around a placeholder leaf. You'd have to change of the tree logic to account for a leaf that's empty but that should appear populated.

I refactored the code however so that the temporary leaf gets replaced by the real one a few lines of code later.

Reply:
Okay, sounds like it's the only reasonable way to do this without a major refactoring.